gobblin-metastore/src/main/java/gobblin/metastore/FsStateStore.java

```diff
@@ -279,9 +276,13 @@ public void createAlias(String storeName, String original, String alias) throws
     }
 
     Path aliasTablePath = new Path(new Path(this.storeRootDir, storeName), alias);
+    Path tmpAliasTablePath = new Path(aliasTablePath.getParent(), new Path(TMP_FILE_PREFIX, aliasTablePath.getName()));
     // Make a copy of the original table as a work-around because
     // Hadoop version 1.2.1 has no support for symlink yet.
-    FileUtil.copy(this.fs, originalTablePath, this.fs, aliasTablePath, false, true, this.conf);
+    HadoopUtils.deleteIfExists(this.fs, tmpAliasTablePath, true);
+    HadoopUtils.copyPath(this.fs, originalTablePath, this.fs, tmpAliasTablePath, this.conf);
+    HadoopUtils.deleteIfExists(this.fs, aliasTablePath, true);
```

Comment:
True. I would rather not have the `StateStore` default to the most recently created file as that violates the contract of the `get(storeName, tableName, stateId)` method which is suppose to either return the `State` associated with `tableName` or `null` if it does not exist.

Will try to think of a safer approach.